src/gnu/prolog/io/TermWriter.java

```diff
@@ -473,7 +473,14 @@ protected void displayAtom(WriteOptions options, AtomTerm atom)
 		if (options.quoted)
 		{
 			String value = atom.value;
-			print(needBeQuoted(value) ? getSingleQuoted(value) : value);
```

Comment:
Yes that is one option that sounds good. Other ideas: change needBeQuoted to take a Term rather than a String or provide a needBeQuoted method that takes a Term and then calls needBeQuoted with the String. Depends how needBeQuoted is used. I would be happy with any of those options.